File: pulumi/k3s-prod-1/main.go
package main

import (
	"k3s-prod-1/monitoring"
	"k3s-prod-1/networking"

	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		if err := monitoring.NewMonitoring(ctx, "prod"); err != nil {
			return err
		}
		if err := networking.NewNetworking(ctx, "prod"); err != nil {
			return err
		}

		return nil
	})
}

File: pulumi/k3s-prod-1/__main__.py
import pulumi
import pulumi_kubernetes as kubernetes


from monitoring import *
from networking import *
from visualization import *


provider = kubernetes.Provider(
    "k3s-prod-1",
    # The name of the kubeconfig context to use
    context="default",
    # Disable server-side apply to make the cluster more reproducible.
    # It will(TODO: not sure) discard any server-side changes to resources when set to false.
    enable_server_side_apply=False,
)

# networking
new_cert_manager = NewCertManager(provider)

# monitoring
new_victoria_metrics = NewVictoriaMetrics(provider)

# visualization
new_kubevirt = NewKubeVirt(provider)
new_virtual_machines = NewVirtualMachines(provider, new_kubevirt)


